nncf/quantization/algorithms/weight_compression/awq.py

```diff
@@ -150,19 +145,30 @@ def apply(
             if not self._backend_entity.is_node_with_weights(nncf_node, graph):
                 continue
 
+            target_node_names = []
             for weight_op_friendly_name, _ in self._backend_entity.get_weight_names_and_port_ids(nncf_node, graph):
                 target_node_names.append(weight_op_friendly_name)
 
-            nncf_node = graph.get_node_by_key(match[0])
-            for weight_op_friendly_name, _ in self._backend_entity.get_weight_names_and_port_ids(nncf_node, graph):
-                merge_node_names.append(weight_op_friendly_name)
-
-            assert len(target_node_names) == len(merge_node_names)
             weight_params = self._all_weight_params[name_mapping[target_node_names[-1]]]
+
             if weight_params.compression_config.num_bits != 4:
                 continue
             target_node = self._nodes_to_compress[name_mapping[target_node_names[-1]]]
-            merge_node = self._nodes_to_compress[name_mapping[merge_node_names[-1]]]
+
+            # avoid matching different patterns for the same node
+            if target_node.node_name in awq_data:
```

Comment:
When can it happen? Which patterns can be applied for the same node? Aren't they mutually exclusive?

Reply:
This code fixes the bug in AWQ with matching multiple patterns for the same node. 
Small unit test would be helpful
![image](https://github.com/openvinotoolkit/nncf/assets/4014476/83a83a79-72fe-449f-b4f9-a9adb9054f02)